### PR TITLE
CFE-942/master: Added test for modules returning nonzero with classes usemodule

### PR DESCRIPTION
**Pull request by nickanderson**

### CODE REVIEW

File: tests/acceptance/02_classes/02_functions/usemodule-returns-nonzero.cf [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+      "foo_usemodule" string =>
+"#!/bin/sh
+echo '=mystring=foo99'
+echo '+class_from_classes_usemodule'
+exit 1
+";
+
+      "foo_commands_module" string =>
+      "#!/bin/sh
+echo '=mystring=foo99'
```
Comment (craigcomstock) [inline tests/acceptance/02_classes/02_functions/usemodule-returns-nonzero.cf]: indent this shebang back to `^` (start of line)?

File: tests/acceptance/02_classes/02_functions/usemodule-returns-nonzero.cf [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        comment => "usemodule seems to return true no matter if the module exists returning 0 or nonzero";
+
+  classes:
+
+      # Since the module exits with non zero, we should not get this class
+
+      "usemodule_expect_no_class_defined_because_return_nonzero"
+        expression => usemodule( "foo-usemodule", "" ),
+        scope => "namespace",
+        if => isexecutable( "$(sys.workdir)/foo-usemodule" );
```
Comment (craigcomstock) [inline tests/acceptance/02_classes/02_functions/usemodule-returns-nonzero.cf]: So if for some reason `$(sys.workdir)/foo-usemodule` is not executable you won't get `Pass` or `FAIL` at the end. Will that cause FAIL I hope?

File: tests/acceptance/02_classes/02_functions/usemodule-returns-nonzero.cf [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+      "Did not expect that a class defined as the result of a classes promise usemodule would be defined if the module used returned nonzero"
+        if => "usemodule_expect_no_class_defined_because_return_nonzero";
+
+      "expected class to be defined based on negative result from usemodule classes promise"
+        if => "usemodule_expect_class_defined_because_return_nonzero";
+
+}
+
+# bundle edit_line append_if_no_line(str)
+# {
```
Comment (craigcomstock) [inline tests/acceptance/02_classes/02_functions/usemodule-returns-nonzero.cf]: Why is all of this commented out stuff here? Isn't it defined in say MPF/lib?

File: tests/acceptance/02_classes/02_functions/usemodule-returns-nonzero.cf (hunk 1 above)
Comment (nickanderson) [reply tests/acceptance/02_classes/02_functions/usemodule-returns-nonzero.cf]: sure i can,

File: tests/acceptance/02_classes/02_functions/usemodule-returns-nonzero.cf (hunk 3 above)
Comment (nickanderson) [reply tests/acceptance/02_classes/02_functions/usemodule-returns-nonzero.cf]: it was just cruft

File: tests/acceptance/02_classes/02_functions/usemodule-returns-nonzero.cf (hunk 2 above)
Comment (nickanderson) [reply tests/acceptance/02_classes/02_functions/usemodule-returns-nonzero.cf]: i hope so also!

**Pull request merged**